src/Analyser/NodeScopeResolver.php

```diff
@@ -1955,13 +1955,11 @@ static function (?Type $offsetType, Type $valueType, bool $optional) use (&$arra
 
 			if (isset($functionReflection) && $functionReflection->getName() === 'shuffle') {
 				$arrayArg = $expr->getArgs()[0]->value;
-				$arrayArgType = $scope->getType($arrayArg);
-
-				if ($arrayArgType instanceof ConstantArrayType) {
-					$arrayArgType = $arrayArgType->getValuesArray()->generalizeToArray();
```

Comment:
fyi this was the one and only public usage of `ConstantArrayType::generalizeToArray()`. I introduced it via 940b386a to address `shuffle()`.